### PR TITLE
new SessionControl library

### PR DESCRIPTION
**Pull request by sopos**
The library is meant to handle multiple sessions and the same time. It has also an ability to handle remote session thus it can be used for orchestration in an controller based multi-host tests environment.

### CODE REVIEW

**Comment by sopos** [comment]
@kkaarreell, check this out :-)

File: SessionControl/README.md [hunk 1]
```diff
@@ -0,0 +1,248 @@
...
+
+# VARIABLES
+
+- **sesID**
+
+    An array holding currently open session IDs. Sessions are strored from index 1,
+    index 0 is always used for the "default" **ID**. The default **ID** is always reset
+    to the last used _ID_.
+
+- **sesRunTimeout**
```
Comment (psss) [inline SessionControl/README.md]: Shall the variables be in uppercase letters? At the first sight this a bit confusing to me.

File: SessionControl/README.md [hunk 2]
```diff
@@ -0,0 +1,248 @@
...
+
+    The input to be send to the session. It may contain also control characters,
+    e.g. **\\003** to send break (^C).
+
+    Note, to execute a command using `sesSend` you need to append **\\r** to confirm
+    it on the prompt.
+
+Returns **0** if successful.
+
+## sesSend
```
Comment (psss) [inline SessionControl/README.md]:
```suggestion
## sesRaw
```

File: SessionControl/README.md [hunk 3]
```diff
@@ -0,0 +1,248 @@
...
+
+Close all the remaining open sessions.
+
+    sesCleanup
+
+Returns **0** if successful. See section ["COMMON RESULT CODE"](#common-result-code) for more details.
+
+# COMMON RESULT CODE
+
+There are specail _RETURN CODES_ commning from the library's functions.
```
Comment (psss) [inline SessionControl/README.md]:
```suggestion
There are special _RETURN CODES_ coming from the library's functions.
```

File: SessionControl/README.md [hunk 4]
```diff
@@ -0,0 +1,248 @@
...
+    end with an error.
+
+- **254**
+
+    The session call timed out. There may be a _command_ hanging or a _pattern_ was
+    not found in time.
+
+- **<254**
+
+    These return codes are typically comming from the executed command.
```
Comment (psss) [inline SessionControl/README.md]:
```suggestion
    These return codes are typically coming from the executed command.
```

File: SessionControl/README.md [hunk 5]
```diff
@@ -0,0 +1,248 @@
...
+
+    Success!
+
+# EXAMPLES
+
+Simply run `whoami` command in a session
+
+    sesOpen
+    sesRun "id"
+    ses Close
```
Comment (psss) [inline SessionControl/README.md]:
```suggestion
    sesClose
```

**Review by psss — COMMENTED** [review]
Looks interesting and could be useful! Have reviewed the docs, skimmed quickly through the rest and found just a couple of minor issues. One comment from the readability point of view: The `ses` prefix seems to me a bit confusing when reading the code, not sure why. Perhaps `session` would be a bit more clear, it's not that many additional characters. But it might be just my personal preference that `sessionOpen`, `sessionRun`, `sessionClose` look better. Maybe one just needs to get used to it. Don't know.

File: SessionControl/README.md (hunk 1 above)
Comment (sopos) [reply SessionControl/README.md]: right, probably `sesRunTIMEOUT` might be better

File: SessionControl/README.md (hunk 2 above)
Comment (sopos) [reply SessionControl/README.md]: good catch

**Comment by sopos** [comment]
`sesRun` vs `sessionRun` is questionable. I'm not a fan of long names/prefixes. It may be good to gather more feedback on this.

Let's vote for each variant in following comments using thumbs up/down.

**Comment by sopos** [comment]
```
sesOpen
rlRun -s "sesRun 'id'"
rlAssertGrep 'root' $rlRun_LOG
sesClose
```

**Comment by sopos** [comment]
```
sessionOpen
rlRun -s "sessionRun 'id'"
rlAssertGrep 'root' $rlRun_LOG
sessionClose
```

**Comment by sopos** [comment]
@psss, @hegerj, @kkaarreell, @milosmalik, please vote above if you have an opinion. Also, feel free to invite others to express their opinions. The more opinions the better.

**Pull request merged**